### PR TITLE
Sqlite3 backend

### PR DESCRIPTION
**Pull request by cp-richard**
I needed SQLite3 support, and we've been using it for some time.  I have tried to cover all the items I think you would want in a good PR.  Doc and unit test are there, but my skills might be lacking in this area as well as my pyproject usage.

Let me know if there are any changes you want or need in the PR to make it easier to review/merge.

### CODE REVIEW

**Comment by DonDebonair** [comment]
Hey @cp-richard this is a great idea! Let me know if you need any help resolving your MyPy issues.

**Comment by cp-richard** [comment]
In the spirit of the redis and dynamodb I was keeping it simple.  I only needed it because I was using the RBAC module (IIRC).  It took me too long to get the mypy/tests/and sqlite connection 'best practices' as it was :smile:

Hopefully it's of some use to someone else and makes my personal fork upstream merges easier :+1:

File: machine/storage/backends/sqlite.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            VALUES (?, ?, ?)
+        """,
+            (key, value, expires),
+        )
+        await self.conn.commit()
+
+    async def get(self, key: str) -> bytes | None:
+        current_GMT = int(time.time())
+        await self.cursor.execute(
+            "SELECT value FROM storage WHERE key=? and (expires > ? OR expires IS NULL)", (key, current_GMT)
```
Comment (DonDebonair) [inline machine/storage/backends/sqlite.py:45]: `expires > ?` will realistically never hold true as `expires` in the `set(...)` method is the number of seconds from now when the key expires.

Example: 

```python
# This sets a key to be expired 300 seconds from now (i.e. 5 minutes)
set("my-key", b"my-value", 300)`
```

This means that in the `get(...)` method, the comparison that is made is `300 > 1691232468` (the latter is the current epoch time gotten from `time.time()`) which will not be true, even if the key was just set.

The change you need to make, is in the `set(...)` method set the `expires` column to `int(time.time()) + expires`. That will make the `get(...)` method work properly.

File: machine/storage/backends/sqlite.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        return row[0] if row else None
+
+    async def delete(self, key: str) -> None:
+        await self.cursor.execute("DELETE FROM storage WHERE key=?", (key,))
+        await self.conn.commit()
+
+    async def has(self, key: str) -> bool:
+        current_GMT = int(time.time())
+        await self.cursor.execute(
+            "SELECT EXISTS(SELECT 1 FROM storage WHERE key=? and (expires > ? OR expires IS NULL))", (key, current_GMT)
```
Comment (DonDebonair) [inline machine/storage/backends/sqlite.py:57]: This will only work if `set(...)` is changed. See comment above.

**Review by DonDebonair — COMMENTED** [review]
There is an error in the _expires_ logic in your code. See comments. If you fix it, it would be nice to provide a test for this. You can do so by mocking/patching `time.time()`

**Review by DonDebonair — CHANGES_REQUESTED** [review]
There is an error in the _expires_ logic in your code. See comments. If you fix it, it would be nice to provide a test for this. You can do so by mocking/patching `time.time()`

**Comment by DonDebonair** [comment]
I wanted to make a few changes, incl. updating the branch with the new stuff on `main` and fixing some variable names. It was easier for me to change them myself than to ask you to do it. Given that I don't have push access to your fork, I created my own PR #844 . You still have commit credits of course.

**Comment by DonDebonair** [comment]
fixed by #844

**Comment by DonDebonair** [comment]
Released in [v0.34.0](https://github.com/DonDebonair/slack-machine/releases/tag/v0.34.0)

**Pull request closed without merging**